tower/src/util/clone_boxed.rs

```diff
@@ -0,0 +1,138 @@
+use super::ServiceExt;
+use futures_util::future::BoxFuture;
+use std::{
+    fmt,
+    task::{Context, Poll},
+};
+use tower_layer::{layer_fn, LayerFn};
+use tower_service::Service;
+
+/// A `Clone + Send` boxed `Service`.
+///
+/// [`CloneBoxService`] turns a service into a trait object, allowing the
+/// response future type to be dynamic, and allowing the service to be cloned.
+///
+/// # Example
+///
+/// ```
+/// use tower::{Service, ServiceBuilder, BoxError, util::CloneBoxService};
+/// use std::time::Duration;
+/// #
+/// # struct Request;
+/// # struct Response;
+/// # impl Response {
+/// #     fn new() -> Self { Self }
+/// # }
+///
+/// // This service has a complex type that is hard to name
+/// let service = ServiceBuilder::new()
+///     .map_request(|req| {
+///         println!("received request");
+///         req
+///     })
+///     .map_response(|res| {
+///         println!("response produced");
+///         res
+///     })
+///     .load_shed()
+///     .concurrency_limit(64)
+///     .timeout(Duration::from_secs(10))
+///     .service_fn(|req: Request| async {
+///         Ok::<_, BoxError>(Response::new())
+///     });
+/// # let service = assert_service(service);
+///
+/// // `CloneService` will erase the type so its nameable
```

Comment:
minor nit: when used as a contraction for "it is", you want "it's", rather than "its":
```suggestion
/// // `CloneService` will erase the type so it's nameable
```

Reply:
One day I'll learn this 😜

Reply:
```suggestion
/// // `CloneBoxService` will erase the type so it's nameable.
```

Reply:
I fixed the typos in https://github.com/tower-rs/tower/pull/615/commits/28915aa9d1cab86aebb38a8fd6f928f0bd1c58d4